### ISSUE TITLE
PyPy's `resolvedPyPyVersion` has a trailing newline

### ISSUE THREAD

**Issue by kurtmckee** [actions/setup-python#609]
**Description:**
While working on #606 I discovered that PyPy's `resolvedPyPyVersion` contains a trailing newline, which causes the `pythonVersion` (passed to `cacheDependencies`) to look like `7.3.11\n-3.9.16`.

**Action version:**
I'm running code against the `main` branch. The bug only exists on Ubuntu and macOS runners. The Windows runner does not have the trailing newline.

**Platform:**
- [x] Ubuntu
- [x] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**
PyPy 7.3.11 with Python 3.9.16

**Repro steps:**  
You can see the issue in the output of the "Installed versions" output group. I've created a repo to demonstrate the issue with a simple workflow:

https://github.com/kurtmckee/demonstrate-pypy-newlines/actions/runs/4151696169/jobs/7182162044

**Expected behavior:**
I expect that the `resolvedPyPyVersion` variable will not have a trailing newline.

**Actual behavior:**
`resolvedPyPyVersion` has a trailing newline, and this can be seen in the "Installed versions" output group's text:

On Ubuntu and macOS:

```
  Successfully set up PyPy 7.3.11
   with Python (3.9.16)
```

On Windows:

```
  Successfully set up PyPy 7.3.11 with Python (3.9.16)
```

**Comment by kurtmckee** [comment]
I'm working to resolve this now and will open a PR shortly.

**Comment by e-korolevskii** [comment]
Hello @kurtmckee! Thank you for reporting this!
We will investigate the issue and get back to you as soon as we have updates.

**Comment by kurtmckee** [comment]
Thanks @e-korolevskii! I think I've fixed this in PR #610. Please take a look when you have an opportunity. 😀

**Comment by dmitry-shibanov** [comment]
Hello @kurtmckee. We've merged your pull request. Could you please try to use setup-python as `actions/setup-python@main` to confirm that everything works as expected? Later we'll prepare the release and update the major tag.

**Comment by kurtmckee** [comment]
Will do! I'll report back shortly.

**Comment by kurtmckee** [comment]
I've tested this against `actions/setup-python@main` and the bug is indeed fixed. Thank you all very much for the quick review and merge!

# Before

[[link]](https://github.com/kurtmckee/demonstrate-pypy-newlines/actions/runs/4151696169/jobs/7182161958#step:2:10)

```text
  Successfully set up PyPy 7.3.11
   with Python (3.9.16)
```

# After

[[link]](https://github.com/kurtmckee/demonstrate-pypy-newlines/actions/runs/4224521614/jobs/7335582282#step:2:10)

```text
  Successfully set up PyPy 7.3.11 with Python (3.9.16)
```

**Comment by dmitry-shibanov** [comment]
Hello @kurtmckee. Thank you for your response. For now I'll reopen the until the new version is released.

**Comment by kurtmckee** [comment]
The fix was released in v4.6.0. Closing.
